sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,681 @@
+                            max_iter=max_iter, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type)
+        assert_warns_message(ConvergenceWarning,
+                             'Initialization %d is not converged. '
+                             'Try different init parameters, '
+                             'or increase n_init, tol '
+                             'or check for degenerate data.'
+                             % max_iter, g.fit, X)
+
+
+def test_gaussian_mixture_n_parameters():
+    # Test that the right number of parameters is estimated
+    rng = np.random.RandomState(0)
+    n_features, n_components = 5, 2
+    n_params = {'spherical': 13, 'diag': 21, 'tied': 26, 'full': 41}
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.n_features = n_features
+        assert_equal(g._n_parameters(), n_params[cv_type])
+
+
+def test_gaussian_mixture_aic_bic():
+    # Test the aic and bic criteria
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 50, 3, 2
+    X = rng.randn(n_samples, n_features)
+    # standard gaussian entropy
+    SGH = 0.5 * (np.log(np.linalg.det(np.cov(X.T, bias=1))) +
+                 n_features * (1 + np.log(2 * np.pi)))
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.fit(X)
+        aic = 2 * n_samples * SGH + 2 * g._n_parameters()
+        bic = (2 * n_samples * SGH +
+               np.log(n_samples) * g._n_parameters())
+        bound = n_features * 3. / np.sqrt(n_samples)
+        assert_true(np.abs(g.aic(X) - aic) / n_samples < bound)
+        assert_true(np.abs(g.bic(X) - bic) / n_samples < bound)
+
+
+def test_gaussian_mixture_verbose():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    n_components = rand_data.n_components
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        g = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=1)
+        h = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=2)
+        k = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=20)
+        old_stdout = sys.stdout
+        sys.stdout = StringIO()
+        try:
+            g.fit(X)
+            h.fit(X)
+            k.fit(X)
+        finally:
+            sys.stdout = old_stdout
+
+
+def test_warm_start():
```

Comment:
This test actually don't test anything, as you can see by changing `max_iter=0` in `h`.
If you plot the data (see below), you can realize that the clusters are very far from each other (the `self.means` are too large in `RandomData`), which makes the convergence complete just after K-means initialization.

```python
import matplotlib.pyplot as plt
plt.plot(X[:, 0], X[:, 1], 'o')
plt.show()
```
You should first reduce the distance between clusters, to have some overlapping.
Also to be sure this test is useful, you should design a failing test, that does not fail with warm start.